docs/docs/how-to/custom-configuration/typescript.md

```diff
@@ -133,6 +193,66 @@ const getServerData: GetServerData<ServerDataProps> = async props => {
 }
 ```
 
+## `gatsby-config.ts`
+
+```ts:title=gatsby-config.ts
+
+import { GatsbyConfig } from "gatsby";
+
+const config: GatsbyConfig = {
+  siteMetadata: {
+    title: "Site Title",
+    siteUrl: "Site URL",
+  },
+  plugins: [],
+};
+
+export default config;
+
+```
+
+## `gatsby-node.ts`
+
+```ts:title=gatsby-node.ts
+
+import { GatsbyNode } from "gatsby";
+
+type Person = {
+  id: number;
+  name: string;
+  age: number;
+};
+
+export const sourceNodes: GatsbyNode["sourceNodes"] = async ({
+  actions,
+  createNodeId,
+}) => {
+  const { createNode } = actions;
+
+  const data = await getSomeData();
+
+  data.forEach((person: Person) => {
+    const node = {
+      ...person,
+      parent: null,
+      children: [],
+      id: createNodeId(`person__${person.id}`),
+      internal: {
+        type: "Person",
+        content: JSON.stringify(person),
+        contentDigest: createContentDigest(person),
+      },
+    };
+
+    createNode(node);
+  });
+};
+```
+
+## Styling
+
+[vanilla-extract](https://vanilla-extract.style/) helps you write type‑safe, locally scoped classes, variables and themes. It's a great solution when it comes to styling in your Typescript project. To use [`vanilla-extract`](https://vanilla-extract.style/), select "vanilla-extract" as your preffered styling solution when initializing your project with with `npm init gatsby`/ `yarn create gatsby`. You can also manually setup your project to use vanilla extract through [gatsby-plugin-vanilla-extract](https://www.gatsbyjs.com/plugins/gatsby-plugin-vanilla-extract/). See [setup instructions](https://www.gatsbyjs.com/plugins/gatsby-plugin-vanilla-extract/)
```

Comment:
We normally only showcase usage with `npm` and skip `yarn`

Links to pages on .com should be relative,  not absolute